draft-ietf-quic-transport.md

```diff
@@ -2859,13 +2859,16 @@ valid frames? -->
 
 ### Sending ACK Frames
 
-An endpoint MUST NOT send more than one packet containing only an ACK frame per
-received packet that contains frames other than ACK and PADDING frames.
-An endpoint MUST NOT send a packet containing only an ACK frame in response
-to a packet containing only ACK or PADDING frames, even if there are packet
-gaps which precede the received packet. This prevents an indefinite feedback
-loop of ACKs. The endpoint MUST however acknowledge packets containing only
-ACK or PADDING frames when sending ACK frames in response to other packets.
+Packets containing only ACK frames are not congestion controlled, so there are
+limits on how frequently they can be sent.  An endpoint MUST NOT send more than
+one packet containing only an ACK frame per received ACK-eliciting packet
+(one containing frames other than ACK and/or PADDING).  An endpoint MUST NOT
+send a packet containing only an ACK frame in response to a non-ACK-eliciting
```

Comment:
So if the stack _really_ wants to ACK some non-ACK-eliciting packet, ACK+PING is ok?  And if ACK+PING is not ok, then ACK+MAX_DATA(repeat latest value) is ok?

My point is that maybe this MUST ought to be replaced with a SHOULD to avoid people having to play these games, when they really want to do so.  (SHOULD still means do it unless you have a great reason not to.)

Reply:
The text about not bundling PING (& friends) with every ACK is for the sender of only-ACKs.

The question here is about the receiver of only-ACKs.  Presumably such endpoint has just sent a train of packets, and when it received an ACK for all the packets, it has nothing more to send.  At that point, it may wish to ACK the last ACK w/o waiting for the receiver to send it ACK+PING.  Would that be sane behavior?  I think so.  But if the sender bundles ACKs for ACKs in packets with data, the receiver will hold at most 1-rtt worth of state, so it is not that bad I guess.

Reply:
What if change the requirement to simply not ACK'ing ACK's too often. That would also do away with PING feedback loops. It seems that we are just moving the problem.

Reply:
The existing text follows what TCP does as closely as I can manage, which I think is the best prior art we can follow here.